### PR TITLE
chore(ci): remove httpie and enhance workflow

### PR DESCRIPTION
**Pull request by mohamedMok**
## I have read [the contributing guidelines](https://mozaic.adeo.cloud/Contributing/)

- [x] Yes
- [ ] No

## Does this PR introduce a [breaking change](https://mozaic.adeo.cloud/Contributing/Developers/GitConventions/#breaking-changes-)?

- [ ] Yes
- [x] No

## Describe the changes

GitHub issue number or Jira issue URL: N/A

### CODE REVIEW

**Review by tiloyi — COMMENTED** [review]
I think that this PR should be separated into two RPs. One PR for updating files specific to Github Actions, and an other PR specific to HTTP => HTTPS redirection.

File: .github/workflows/push.yml [hunk 1]
```diff
@@ -30,17 +32,27 @@ jobs:
...
           GCLOUD_AUTH: ${{ secrets.GCLOUD_AUTH }}
         with:
-          args: ci:wait $(echo ${{ github.ref }} | iconv -t ASCII | sed -r 's/[^a-zA-Z0-9]+//g' | sed -r 's/refsheads//g' | sed -r 's/^-+\\|-+$//g' | tr A-Z a-z)
+          args: ci:wait ${{ env.GITHUB_EVENT_REF_SLUG_URL }}
+      - name: deployment
+        uses: bobheadxi/deployments@d214e9228f80adbf4cc2f6e470f2e66ebac5a1b3
+        id: deployment
+        with:
+          step: start
+          token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (tiloyi) [inline .github/workflows/push.yml:41]: Maybe it's better to keep using the `${{ secrets.SVC_GH_API_TOKEN }}` ?

File: .github/workflows/push.yml [hunk 2]
```diff
@@ -30,17 +32,27 @@ jobs:
...
-          GH_API_TOKEN: ${{ secrets.SVC_GH_API_TOKEN }}
+          args: app deploy app.demo.yaml --project=mozaic-design-system-dtrp --version=${{ env.GITHUB_EVENT_REF_SLUG_URL }} --no-promote
+      - name: Update deployment status
+        uses: bobheadxi/deployments@d214e9228f80adbf4cc2f6e470f2e66ebac5a1b3
         with:
-          args:
-            POST https://535e8ft89a.execute-api.eu-west-3.amazonaws.com/dev/deployment
-            "ref=${{ github.ref }} env=staging gh_token=$GH_API_TOKEN"
+          step: finish
+          token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (tiloyi) [inline .github/workflows/push.yml:55]: Maybe it's better to keep using the `${{ secrets.SVC_GH_API_TOKEN }}` ?

File: .github/workflows/push.yml [hunk 3]
```diff
@@ -30,17 +32,27 @@ jobs:
...
+          status: ${{ job.status }}
+          deployment_id: ${{ steps.deployment.outputs.deployment_id }}
+          env_url: https://${{ env.GITHUB_EVENT_REF_SLUG_URL }}-dot-demo-dot-mozaic-design-system-dtrp.ey.r.appspot.com/
```
[the remark this answers is not in the document]
Comment (mohamedMok) [reply .github/workflows/push.yml:58]: I'll test it out

File: app.yaml [hunk 4]
```diff
@@ -4,12 +4,18 @@ instance_class: F1
```
[the remark this answers is not in the document]
Comment (mohamedMok) [reply app.yaml]: make sense i'll remove them

File: .github/workflows/push.yml [hunk 5]
```diff
@@ -30,17 +34,27 @@ jobs:
...
+          ref: ${{ github.head_ref }}
+          transient: true
+      - uses: google-github-actions/deploy-appengine@main
+        with:
+          project_id: mozaic-design-system-dtrp
+          credentials: ${{ secrets.GCLOUD_AUTH }}
+          version: ${{ steps.getbranch.outputs.branch }}
+          deliverables: app.demo.yaml
+      - name: Update deployment status
+        uses: bobheadxi/deployments@v0.5.2
```
Comment (tiloyi) [inline .github/workflows/push.yml:54]:
```
- name: Deploy on App Engine
  id: appengine
  uses: google-github-actions/deploy-appengine@main
  with:
    project_id: mozaic-design-system-dtrp
    credentials: ${{ secrets.GCLOUD_AUTH }}
    version: ${{ steps.getbranch.outputs.branch }}
    deliverables: app.demo.yaml
    promote: false
```

File: .github/workflows/push.yml [hunk 6]
```diff
@@ -30,17 +34,27 @@ jobs:
...
         with:
-          args:
-            POST https://535e8ft89a.execute-api.eu-west-3.amazonaws.com/dev/deployment
-            "ref=${{ github.ref }} env=staging gh_token=$GH_API_TOKEN"
+          step: finish
+          token: ${{ secrets.SVC_GH_API_TOKEN }}
+          status: ${{ job.status }}
+          deployment_id: ${{ steps.deployment.outputs.deployment_id }}
+          env_url: https://${{ steps.getbranch.outputs.branch }}-dot-demo-dot-mozaic-design-system-dtrp.ey.r.appspot.com/
```
Comment (tiloyi) [inline .github/workflows/push.yml:60]: ~~env_url: https://${{ steps.getbranch.outputs.branch }}-dot-demo-dot-mozaic-design-system-dtrp.ey.r.appspot.com/~~ 

```
env_url: ${{ steps.appengine.outputs.url }}
```

File: .github/workflows/release.yml [hunk 7]
```diff
@@ -7,29 +7,52 @@ jobs:
...
+          env: production
+          no_override: false
+          ref: ${{ github.head_ref }}
+
+      - uses: google-github-actions/deploy-appengine@main
+        with:
+          project_id: mozaic-design-system-dtrp
+          credentials: ${{ secrets.GCLOUD_AUTH }}
+          version: ${{ env.GITHUB_EVENT_REF_SLUG_URL }}
+          deliverables: app.demo.yaml
```
Comment (tiloyi) [inline .github/workflows/release.yml:40]: It seems to me that lines 35 to 40 are not necessary.

File: .github/workflows/release.yml [hunk 8]
```diff
@@ -7,29 +7,52 @@ jobs:
...
+
       - name: Deploy Tag
-        uses: actions-hub/gcloud@master
+        uses: agoogle-github-actions/deploy-appengine@main
```
Comment (tiloyi) [inline .github/workflows/release.yml:43]: ~~agoogle-github-actions/deploy-appengine@main~~ => google-github-actions/deploy-appengine@main

File: .github/workflows/release.yml [hunk 9]
```diff
@@ -7,29 +7,52 @@ jobs:
...
         env:
-          PROJECT_ID: mozaic-design-system-dtrp
-          APPLICATION_CREDENTIALS: ${{ secrets.GCLOUD_AUTH }}
+          project_id: mozaic-design-system-dtrp
+          credentials: ${{ secrets.GCLOUD_AUTH }}
+          version: ${{ steps.gettag.outputs.branch }}
+          deliverables: app.yaml
         with:
           args: app deploy app.yaml --project=mozaic-design-system-dtrp --version=$(echo ${{ github.ref }} | iconv -t ASCII | sed -r 's/[^a-zA-Z0-9]+//g' | sed -r 's/refstags//g' | sed -r 's/^-+\\|-+$//g' | tr A-Z a-z) --promote
```
Comment (tiloyi) [inline .github/workflows/release.yml:50]: Please replace lines 44 to 50, by : 

```
  with:
    project_id: mozaic-design-system-dtrp
    credentials: ${{ secrets.GCLOUD_AUTH }}
    version: ${{ steps.gettag.outputs.branch }}
    deliverables: app.yaml
    promote: true
```

**Review by tiloyi — APPROVED** [review]

**Pull request merged**